riscv_config/schemas/schema_custom.yaml

```diff
@@ -0,0 +1,185 @@
+          schema:
+            fields:
+              type: list
+              default: []
+            keysrules:
+              type: dict
+              regex: '[a-zA_Z0-9_]*'
+              schema:
+                description:
+                  type: string
+                  default: subfield.
+                shadow:
+                  type: string
+                  default:
+                  nullable: True
+                shadow_type:
+                  type: string
+                  default: rw
+                  nullable: True
+                  allowed: ['rw', 'ro']
+                msb:
+                  type: integer
+                lsb:
+                  type: integer
+                implemented:
+                  type: boolean
+                  default: false
+                type:
+                  type: dict
+                  oneof:
+                    - schema:
+                        ro_constant:
+                          type: integer
+                    - schema:
+                        ro_variable:
+                          type: boolean
+                    - schema:
+                        warl:
+                          type: dict
+                          schema:
+                            dependency_fields:
+                              type: list
+                              empty: true
+                              schema:
+                                type: string
+                                nullable: true
+                            legal:
+                              type: list
+                              schema:
+                                type: string
+                            wr_illegal:
+                              type: list
+                              schema:
+                                type: string
+    uarch_signals:
+      required: false
+      type: dict
+      schema:
+        keysrules:
+          regex: '^uarch_'
+          type: dict
+          oneof:
+          - schema:
+              msb:
+                type: integer
+                required: true
+                min: 0
+                max: 62
+              lsb:
+                type: integer
+                required: true
+                min: 1
+                max: 63
+              reset-val:
+                type: integer
+                required: true
+                min: 0
+                max: 0xFFFFFFFFFFFFFFFF
+              legal:
+                type: list
+                required: false
+                nullable: true
+                schema:
+                  type: integer
+          - schema:
+              subfields:
+                type: dict
+                schema:
+                  msb:
+                    type: integer
+                    required: true
+                    min: 0
+                    max: 62
+                  lsb:
+                    type: integer
+                    required: true
+                    min: 1
```

Comment:
min of lsb cant be greater than msb. Additionally, the way schema is defined, each subfield will have to be a minimum of 2 bits wide. No min-max bounds are necessary for these registers. The schema would also benefit from an additional check_with function to ensure that the subfields are non-overlapping.